libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/MapRouteLine.kt

```diff
@@ -1588,6 +1530,159 @@ internal class MapRouteLine(
                 else -> y2
             }
         }
+
+        /**
+         * This will extract all of the road classes for the various sections of the route
+         * and determine the distance from the origin point to the starting point for each
+         * road class section and associate that data with a traffic congestion value.
+         *
+         * Each item returned represents a distance from the origin point, the road class
+         * beginning at that distance and the traffic congestion value for that section. The
+         * road class and congestion remain until the next item in the collections.
+         *
+         * The items returned are ordered from the point closest to the origin to the point
+         * farthest from the origin. In other words from the beginning of the route until the end.
+         *
+         * @param route the the calculations should be performed on.
+         */
+        fun getRouteLineTrafficExpressionData(route: DirectionsRoute):
+            List<RouteLineTrafficExpressionData> {
+                val distances = route.legs()?.mapNotNull {
+                    it.annotation()?.distance()
+                }?.flatten()
+                ifNonNull(distances) { distanceList ->
+                    val roadClassMap = mutableMapOf<Int, String>()
+                    route.legs()?.asSequence()
+                        ?.mapNotNull { it.steps() }
+                        ?.flatten()
+                        ?.mapNotNull { it.intersections() }
+                        ?.flatten()
+                        ?.filter {
+                            it.geometryIndex() != null
+                        }?.toList()?.forEach {
+                            roadClassMap[it.geometryIndex()!!] =
+                                it.mapboxStreetsV8()?.roadClass() ?: "unknown"
```

Comment:
```suggestion
                                it.mapboxStreetsV8()?.roadClass() ?: "intersection_without_road_class_fallback"
```

Let's fall back to something that definitely won't make it into the actual response in the future.